### PR TITLE
allow use of all write() variants

### PR DESCRIPTION
**Pull request by mzero**
Adafruit_USBD_MIDI implements Stream, and transitively Print. 

Print overloads write() with several versions, only one of which must be implemented, write(uint8_t).

Because Adafruit_USBD_MIDI declares write(uint8_t) in its class definition (as it must), it will _lose_ the overloaded versions, unless explicitly including them with a using declaration... Which is what this commit does.

Without this using declaration, users of this class cannot call write(const char* str) or write(const uint8_t *, size_t) as they would expect, since it inherits from Print.

### CODE REVIEW

**Review by hathach — COMMENTED** [review]
Ah thanks, I definitely miss this one. However, I see more of the 
`using Print::write;` than the Stream one. I am sure there is little difference, maybe we should use the Print::write like others.

**Comment by mzero** [comment]
_tl;dr:_ It is long term safer to expose overloaded members from direct base classes, rather than indirect base classes. Hence `using Stream::write;`.

----

At present, `Print` defines four, overloaded `write()` members, and `Stream` has none. In this state, `using Print::write;` and `using Stream::write;` have the same effect... because `Stream` inherits, and exposes all of `Print`'s `write()` members -- since it defines none of it's own.

If `Stream` were to someday add a `write()` method, it would need to also add `using Print::write;` to ensure that `Print`'s version continue to be exposed.  At this point, code that derived from `Stream`, but had used `using Print::write;` would now find that they'd have to be changed to `using Stream::write;`. Had that code used `using Stream::write;` all along, it would not have had to change.

Hence, it is generally cleanest to refer to the direct base classes: Code that derives from `Stream`, if it declares `using Stream::write;` will work now, and in the future, should `Stream` see fit to overload `write()`.

---

**Another way of thinking about this:** A derived class saying it wishes to expose overloaded methods that it inherits from its base class.  Since it is inheriting from `Stream`, it should refer to that, and expose what `Stream` wishes to expose. That `Stream` currently exposes `write()` from `Print` is its business... and should it change what is exposed (generally by adding more overloads), then it is up to `Stream` to ensure they are all properly exposed.   Classes deriving from `Stream`, should simply use `using Stream::write;`.

**Comment by mzero** [comment]
Here is some minimal code to illustrate the idea:
```C++

class Print {
public:
  virtual void write(int);
  virtual void write(const char*, int);
  virtual void write(char) = 0;
};

class Stream : public Print {
public:
  // using Print::write;            /* 1 */
  // void write(float, bool);
};

class Derived : public Stream {
public:
  using Stream::write;              /* 3 */
  void write(char) { }
};

void foo() {
  Derived x;
  x.write("abc", 3);
  // x.write(1.2, false);           /* 2 */
}
```
1. Uncommenting these lines requires no change in `Derived`, and `foo()` still compiles.
2. Uncommenting this line shows that `foo()` now has access to the new overload in `Stream`.
3. Had this line used `Print`, while the original version of `foo()` would have compiled, Once `Stream` added the new overload, `Derived` would have to change this line to use `Stream;` for `foo()` to have access.

Hence, easiest for `Derived` to refere to `Stream::write` all along.

**Review by hathach — APPROVED** [review]
Thank you very much, the explanation it is much more detail than I would need. Approved !!!

**Comment by mzero** [comment]
True - I'm kinda verbose (just ask my husband!) ... But, this way, the thinking behind the choice is documented - and perhaps it will help others in the future.

Besides, it was fun to develop the minimal clear case, and show how it works. Thanks for review and approval.

**Comment by hathach** [comment]
> True - I'm kinda verbose (just ask my husband!) ... But, this way, the thinking behind the choice is documented - and perhaps it will help others in the future.
> 
> Besides, it was fun to develop the minimal clear case, and show how it works. Thanks for review and approval.

we are the one to say thanks. It is hard to find someone verbal in development world. It is much easier to talk less or didn't give out the explanation. Very well thought out solution.
